UI/MapWnd.cpp

```diff
@@ -6985,9 +6988,51 @@ void MapWnd::DispatchFleetsExploring() {
                 auto pair = GetPathfinder()->ShortestPath(
                     fleet->SystemID(), system_supply.first, empire_id);
 
+                auto route = pair.first;
+
+                // check if path contains hostile ships/fleets
+                bool is_unobstructed = true;
+                if (!ignore_hostile_route) {
+                    // TODO: replace with a general purpose universe, empire, or pathfinder function
+                    for (auto next_system = ++(route.begin()); next_system != route.end(); ++next_system) {
+                        for (auto object_id : GetSystem(*next_system)->ContainedObjectIDs()) {
+                            auto object = GetUniverseObject(object_id);
+                            if (!object) {
+                                ErrorLogger() << "Unable to find object with id: " << object_id;
+                                continue;
+                            }
+
+                            // Only ships/fleets are of concern to exploring ships
+                            auto object_type = object->ObjectType();
+                            if (object_type != OBJ_SHIP && object_type != OBJ_FLEET)
+                                continue;
+
+                            // Owned by non-hostile empire
+                            if (object->Owner() != ALL_EMPIRES &&
+                                Empires().GetDiplomaticStatus(empire_id, object->Owner()) != DIPLO_WAR)
+                            { continue; }
+
+                            auto object_vis_map = GetUniverse().GetObjectVisibilityTurnMapByEmpire(object_id, empire_id);
+                            auto object_vis_it = object_vis_map.find(VIS_BASIC_VISIBILITY);
+                            if (object_vis_it != object_vis_map.end()) {
+                                TraceLogger() << "obstructed path at system " << *next_system << " by object " << object_id;
+                                is_unobstructed = false;
+                                break;
+                            }
+                        }
+                        if (!is_unobstructed)
+                            break;
+                    }
+                }
+                if (!is_unobstructed) {
+                    std::string obstructed_route_msg("Exploring ship not using obstructed route:");
+                    for (auto route_system_id : route)
+                        obstructed_route_msg += " " + std::to_string(route_system_id);
```

Comment:
could just output "Route: 2 24 82 58 173" or similar, rather than the "Exploring ship..." then a single number repeatedly over several lines